src/Collection/ImmArray.php

```diff
@@ -484,4 +484,36 @@ protected function arraySort(callable $cb = null)
         }
         return static::fromArray($ar);
     }
+    
+    /**
+     * Checks if element exists in collection
+     *
+     * @param $element
+     * @param bool $strict use strict comparison?
+     * @return bool
+     */
+    public function has($element, $strict = false)
+    {
+        return \in_array($element, $this->toArray(), strict);
```

Comment:
We avoid unnecessary conversions to array - it doesn't always happen in-place and can result in very large objects being created.

```
foreach ($this as $el) {
  if ($el === $element) return true;
}
return false
```

Reply:
done